src/libraries/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -488,6 +489,39 @@ public static IEnumerable<string> ReadLines(string path, Encoding encoding)
             return ReadLinesIterator.CreateIterator(path, encoding);
         }
 
+        public static IAsyncEnumerable<string> ReadLinesAsync(string path, CancellationToken cancellationToken = default)
+        {
+            return ReadLinesAsync(path, Encoding.UTF8, cancellationToken);
+        }
+
+        public static IAsyncEnumerable<string> ReadLinesAsync(string path, Encoding encoding, CancellationToken cancellationToken = default)
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+            if (path.Length == 0)
+                throw new ArgumentException(SR.Argument_EmptyPath, nameof(path));
+
+            return InternalReadLinesAsync(path, encoding, cancellationToken);
+        }
+
+        private static async IAsyncEnumerable<string> InternalReadLinesAsync(string path, Encoding encoding, [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+            Debug.Assert(encoding != null);
+
+            using StreamReader sr = AsyncStreamReader(path, encoding);
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string? line;
+            while ((line = await sr.ReadLineAsync().ConfigureAwait(false)) != null)
```

Comment:
The lack of cancellation support in ReadLineAsync is unfortunate for this API in particular, as it means the cancellation token won't help unblocking work stuck because no data is available yet (e.g. if reading from a NetworkStream).  We should strongly consider implementing https://github.com/dotnet/runtime/issues/20824 before adding this API that would sit on top of it. cc: @terrajobst, @carlossanlop, @Jozkee

Reply:
Yeah, that would be nice. I have no problem holding off until #20824 gets in 👍🏻